File: backend/src/main/java/com/bookbook/domain/report/dto/response/ReportSimpleResponseDto.java
package com.bookbook.domain.report.dto.response;


import com.bookbook.domain.report.entity.Report;
import com.bookbook.domain.report.enums.ReportStatus;
import lombok.Builder;
import lombok.NonNull;

import java.time.LocalDateTime;

@Builder
public record ReportSimpleResponseDto(
        @NonNull Long id,
        @NonNull ReportStatus status,
        @NonNull Long reporterUserId,
        @NonNull Long targetUserId,
        @NonNull LocalDateTime createdDate
){
    public static ReportSimpleResponseDto from(Report report){
        return ReportSimpleResponseDto.builder()
                .id(report.getId())
                .status(report.getStatus())
                .reporterUserId(report.getReporterUser().getId())
                .targetUserId(report.getTargetUser().getId())
                .createdDate(report.getCreatedDate())
                .build();
    }
}

File: backend/src/main/java/com/bookbook/domain/report/dto/response/ReportSimpleResponseDto.kt
package com.bookbook.domain.report.dto.response

import com.bookbook.domain.report.entity.Report
import com.bookbook.domain.report.enums.ReportStatus
import java.time.LocalDateTime

data class ReportSimpleResponseDto(
    val id: Long,
    val status: ReportStatus,
    val reporterUserId: Long,
    val targetUserId: Long,
    val createdDate: LocalDateTime
) {
    constructor(report: Report) : this(
        id = report.id,
        status = report.status,
        reporterUserId = report.reporterUserId,
        targetUserId = report.targetUserId,
        createdDate = report.createdDate
    )
}